backends/ebpf/ebpfControl.cpp

```diff
@@ -47,21 +47,65 @@ bool ControlBodyTranslator::preorder(const IR::PathExpression* expression) {
 }
 
 void ControlBodyTranslator::processFunction(const P4::ExternFunction* function) {
-    ::error("%1%: Not supported", function->method);
+    if (function->method->name.name == "assert") {
+        BUG_CHECK(function->expr->arguments->size() == 1,
+                "%1%: Expected one argument in ", function);
+        builder->emitIndent();
+        builder->append("if (");
+        visit(function->expr->arguments->at(0)->expression);
+        builder->append(" == false) ");
+        builder->blockStart();
+        builder->emitIndent();
+        builder->appendFormat("% s = % s;",
+             control->program->errorVar.c_str(), P4::P4CoreLibrary::instance.assertError);
+        builder->newline();
+        auto srcInfo = function->expr->srcInfo;
+        auto sourceFragment = srcInfo.toBriefSourceFragment();
+        auto sourceFile = srcInfo.getSourceFile();
+        auto linePosition = srcInfo.toPositionString();
+        builder->emitIndent();
+        builder->appendFormat(
+            "fprintf(stderr, \"Assertion error: ' % s ' failed, file ' % s ', line % s \\n\");",
+            sourceFragment.c_str(), sourceFile.c_str(), linePosition.c_str());
+        builder->newline();
+        builder->emitIndent();
+        builder->appendFormat("goto %s;", IR::ParserState::reject.c_str());
+        builder->newline();
+        builder->blockEnd(true);
+        return;
+    } else if (function->method->name.name == "assume") {
```

Comment:
this code also has a lot of duplication.
what is the difference between assert and assume really?

Reply:
I thought that assert reports error when expr == false and assume reports it when expr == true. 
From comments in [behavioral_model pull request](https://github.com/p4lang/behavioral-model/pull/737/commits/9846038dfc2d9b98dc10e4e679a8b3c3efe333ea) I see now that my assumption was wrong.
As I understood from the comments, both of them should report error when expr == false, but now I have some doubts about their differences in usage. If they should have similar behavior, what is the real difference between them?

Reply:
The intended difference is in how formal proof tools use them, not in how behavioral-model uses them.  I have tried to explain the difference in the original issue here: https://github.com/p4lang/p4c/issues/1548  If that does not make it clear, please consider asking a question in that issue's comments (or here).

Reply:
The question remains: how should the bmv2 and ebpf back-ends implement these functions?
Are these compile-time only - should we always remove them?

Reply:
I have tried to understand more from the original issue, but I still do not comprehend the differences that you mentioned. @jafingerhut  could you give me an example that will help me to better understand it, please?